File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int n = nums1.length;
        int m = nums2.length;
        int[] nums = new int[n+m];
        int i = 0, j = 0, k = 0;
        while(i<n && j<m){
            if(nums1[i]<nums2[j]){
                nums[k] = nums1[i];
                i++;  
            }else{
                nums[k] = nums2[j];
                j++;
            }
            k++;

        }

        while(i<n){
            nums[k] = nums1[i];
            i++;
            k++;
        }
        while(j<m){
              nums[k] = nums2[j];
              j++;
              k++;
              }
           
  
        double med ;
        int s = (n+m)/2;
        if((n+m)%2 == 0){
            med = (double) (nums[s]+nums[s-1]) / 2.0;
            
        }else{
            med =  nums[s];
        }
        return med;
        
    }
}

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        n, m = len(nums1), len(nums2)
        i = j = 0
        r = []
        while i<n and j<m:
            if nums1[i] < nums2[j]:
                r.append(nums1[i])
                i += 1
            else:
                r.append(nums2[j])
                j += 1
        r.extend(nums1[i:])
        r.extend(nums2[j:])
        t = n + m
        mid = t//2
        if t%2==0:
            return (r[mid] + r[mid-1])/2
        else:
            return r[mid]
        
        

            


        